culinary-canvas/capstone-project/frontend/my_capstone_project/src/UserProfile.jsx

```diff
@@ -8,13 +8,65 @@ import SavedRecipes from "./SavedRecipes";
 import ProfilePictureModal from "./ProfilePictureModal";
 import DietaryRestrictions from "./DietaryRestrictions";
 import MealPlannerDashboard from "./MealPlannerDashboard";
+import BarChart from "./BarChart";
+import { getSocket } from "./socket";
+import RateLimitModal from "./RateLimitModal";
 
+const backendAddress = import.meta.env.VITE_BACKEND_ADDRESS;
 function UserProfile() {
-  const { user, logOut } = useAuth();
-  const navigate = useNavigate();
+  const [healthScores, setHealthScores] = useState([]);
+  const { isAuthenticated, user, logOut } = useAuth();
   const [isModalOpen, setIsModalOpen] = useState(false);
-  const backendAddress = import.meta.env.VITE_BACKEND_ADDRESS;
+  const [isRateLimitModalOpen, setRateLimitModalOpen] = useState(false);
+  const navigate = useNavigate();
 
+  useEffect(()=> {
+    const socket = getSocket();
+    if (isAuthenticated && user){
+      const fetchHealthScores = async() => {
+        try{
+          const response = await fetch(`${backendAddress}/meal-planner-health-scores`, {
+            credentials: 'include',
+          });
+          if (response.status === 429){
+            setRateLimitModalOpen(true);
+            setTimeout(() => setRateLimitModalOpen(false), 120000)
+          }
+          else{
+            const data = await response.json();
+            setHealthScores(data)
+          }
+        }catch(error){
+          console.error('Error fetching health scores: ', error)
+        }
+      };
+      fetchHealthScores();
+      socket.on('connect', () => {
+        socket.emit('suscribe', user)
+    });
+
+      socket.on('healthScoresUpdated', (updatedHealthScores) => {
```

Comment:
Indenting is a bit off here. If it's possible to use an auto-formatter, definitely give it a shot! It helps with code logic readability.